docs/ADR-Resize-Channel-Security.md

```diff
@@ -0,0 +1,130 @@
+5. Result: User has committed 20 USDC despite only having 10 USDC in their balance
+
+## Decision Drivers
+* Security: Prevent double-spending across multiple chains
+* User Experience: Maintain a seamless resize process
+* System Integrity: Ensure balance accounting remains accurate
+* Technical Feasibility: Implement with minimal changes to existing contracts
+* Operational Overhead: Consider the impact on server operations
+
+## Considered Options
+
+### Option 1: Timeout-Based Fund Locking
+* Lock funds in the database when a resize is requested
+* Set a timeout to release locks if not confirmed on-chain
+* Challenge: The smart contract has no notion of state expiration, so users could still submit expired states
+
+### Option 2: State Versioning with Registry
+* Maintain a server-side registry of valid state versions
+* When signing a new state, include a global version number in state data
+* When processing on-chain events, invalidate all previous versions
+* Challenge: **Critical flaw** - Once a state is signed by the server, it can be submitted to the blockchain regardless of database state. Without smart contract modifications to validate against the external registry, this approach only detects double-spending after it occurs but doesn't prevent it.
+
+### Option 3: Cross-Chain Lock Commitments
+* Modify smart contracts to require a "lock" transaction on a primary chain
+* This lock creates a verifiable on-chain commitment of total funds allocation
+* Other chains verify against this commitment before processing resizes
+* Each resize reduces the available commitment amount
+* Challenge: Requires smart contract modifications, adds cross-chain dependencies, and significantly complicates the user experience with additional transactions
+
+### Option 4: Server-Side Blockchain Submission
+* Instead of returning signed states to users, server directly submits resize transactions
+* Server tracks on-chain confirmations before allowing additional resizes
+* Server manages transaction queuing, gas prices, and retries
+
+## Decision
+**We want to implement Option 4: Server-Side Blockchain Submission for channel resize operations.**
+
+This approach offers the strongest security guarantees while maintaining a good user experience. The server will directly submit resize transactions to the blockchain after validating user requests, eliminating the window for double-spending.
+
+After thorough analysis, we determined that:
+- Option 1 (Timeout-Based Fund Locking) fails because signed states remain valid indefinitely
+- Option 2 (State Versioning) cannot prevent double-spending without smart contract modifications
+- Option 3 (Cross-Chain Lock Commitments) would work but requires significant contract changes and degrades UX
+
+Server-side submission (Option 4) addresses the vulnerability without contract modifications by ensuring users never receive multiple valid states they could submit independently.
```

Comment:
I think this should be reevaluated according to my [this](https://github.com/erc7824/nitrolite/pull/149/files#r2125941563) and [this](https://github.com/erc7824/nitrolite/pull/149/files#r2125963442) comment.